docs/user-guide/deployments.md

```diff
@@ -454,6 +454,142 @@ nginx-deployment-3066724191   0         0         1h
 Note: You cannot rollback a paused Deployment until you resume it.
 
 
+## Failed Deployment
+
+Your Deployment may get stuck trying to deploy its newest ReplicaSet without ever completing, due to:
+
+* insufficient quota
+
+* readiness probe failures
+
+* image pull error
+
+* insufficient permissions
+
+* limit ranges
+
+* application runtime misconfiguration
+
+In those cases you can specify a timeout parameter in the spec ([spec.progressDeadlineSeconds](#progress-deadline-seconds))
+that denotes the seconds you want to wait for your Deployment to report any progress. Once the deadline is exceeded, meaning
+your Deployment will not report any progress for more than progressDeadlineSeconds, the deployment controller will add a
+condition with `Type=Progressing`, `Status=False`, and `Reason=ProgressDeadlineExceeded` in the status of the Deployment.
+For more information on status conditions, see [here](https://github.com/kubernetes/kubernetes/blob/{{page.githubbranch}}/docs/devel/api-conventions.md#typical-status-properties). 
+Note that at the moment, Kubernetes will do nothing more than just report with a condition with `Reason=ProgressDeadlineExceeded`.
+
+Another thing to note is transient errors eg. your rollout times out and a condition with `Reason=ProgressDeadlineExceeded`
```

Comment:
I think you're trying to say is something like this:

"If you're experiencing transient errors or failures with your Deployments, such as a timeout error during rollout that results in the condition `Reason=ProgressDeadlineExceeded` being set, you may have insufficient quota available. You can address an issue of insufficient quota by scaling down your Deployment, by scaling down other controllers you might be running, or by increasing quota in your namespace."